api/src/main/java/jakarta/enterprise/invoke/InvokerBuilder.java

```diff
@@ -0,0 +1,369 @@
+ *         return transformedResult;
+ *     }
+ * }
+ * </pre>
+ *
+ * The caller of this invoker may pass {@code null} as the target instance, because
+ * the invoker will lookup the target instance on its own. Therefore, calling
+ * {@code invoker.invoke(null, new Object[] {"world"})} will return
+ * {@code "Hello WORLD! Hello WORLD!"}.
+ *
+ * <h2>General requirements</h2>
+ *
+ * To refer to a transformer or a wrapper, all methods in this builder accept:
+ * 1. the {@code Class} that that declares the method, and 2. the {@code String} name
+ * of the method.
+ * <p>
+ * Transformers may be {@code static}, in which case they must be declared directly
+ * on the given class, or they may be instance methods, in which case they may be declared
+ * on the given class or inherited from any of its supertypes.
+ * <p>
+ * It is possible to register only one transformer of each kind, or for each argument
+ * position in case of argument transformers. Attempting to register a second transformer
+ * of the same kind, or for the same argument position, leads to an exception.
+ * <p>
+ * Wrappers must be {@code static} and must be declared directly on the given class.
+ * It is possible to register only one wrapper. Attempting to register a second wrapper
+ * leads to an exception.
+ * <p>
+ * It is a deployment problem if no method with given name and valid signature is found,
+ * or if multiple methods with given name and different valid signatures are found. It is
+ * a deployment problem if a registered transformer or wrapper is not {@code public}.
+ * <p>
+ * Transformers and wrappers may declare the {@code throws} clause. The declared exception
+ * types are ignored when searching for the method.
+ * <p>
+ * For the purpose of the specification of transformers and wrappers below, the term
+ * <em>any-type</em> is recursively defined as: the {@code java.lang.Object} class type,
+ * or a type variable that has no bound, or a type variable whose first bound is
+ * <em>any-type</em>.
+ *
+ * <h2>Input lookups</h2>
+ *
+ * For the target instance and for each argument, it is possible to specify that the value
+ * passed to {@code Invoker.invoke()} should be ignored and a value should be looked up
+ * from the CDI container instead.
+ * <p>
+ * For the target instance, a CDI lookup is performed with the required type equal to the bean
+ * class of the bean to which the target method belongs, and required qualifiers equal to the set
+ * of all qualifier annotations present on the bean class of the bean to which the target method
+ * belongs. When the target method is {@code static}, the target instance lookup is skipped.
+ * <p>
+ * For an argument, a CDI lookup is performed with the required type equal to the type of
+ * the corresponding parameter of the target method, and required qualifiers equal to the set
+ * of all qualifier annotations present on the corresponding parameter of the target method.
+ * <p>
```

Comment:
At some point, we probably need to be more specific about what we mean by a CDI lookup.

I guess at deployment we're performing typesafe resolution, and then when the method is invoked we're obtaining a contextual instance (or contextual reference?) of the Bean?

Reply:
It would be a contextual reference. But I think I'd like to keep that specified in terms of [programmatic lookup](https://jakarta.ee/specifications/cdi/4.0/jakarta-cdi-spec-4.0.html#programmatic_lookup), because that shields us from these low-level concerns.